llvm/include/llvm/Analysis/MemoryLocation.h

```diff
@@ -64,16 +64,19 @@ class Value;
 //
 // If asked to represent a pathologically large value, this will degrade to
 // std::nullopt.
+// Store Scalable information in bit 62 of Value. Scalable information is
+// required to do Alias Analysis on Scalable quantities
 class LocationSize {
   enum : uint64_t {
     BeforeOrAfterPointer = ~uint64_t(0),
-    AfterPointer = BeforeOrAfterPointer - 1,
-    MapEmpty = BeforeOrAfterPointer - 2,
-    MapTombstone = BeforeOrAfterPointer - 3,
+    ScalableBit = uint64_t(1) << 62,
+    AfterPointer = (BeforeOrAfterPointer - 1) & ~ScalableBit,
+    MapEmpty = (BeforeOrAfterPointer - 2) & ~ScalableBit,
+    MapTombstone = (BeforeOrAfterPointer - 3) & ~ScalableBit,
```

Comment:
Do we need to exclude the ScalableBit for these? Doesn't look relevant.

Reply:
it seems that it's needed, else some tests like `llvm/test/Analysis/BasicAA/libfuncs.ll` fails

Reply:
Hm, how does it fail?

I guess ScalableBit could matter for AfterPointer, but it shouldn't matter for MapEmpty or MapTombstone.